pkg/resolvers/endpoints.go

```diff
@@ -163,8 +169,12 @@ func (r *defaultEndpointsResolver) resolveNetworkPeers(ctx context.Context, poli
 			}
 			var portList []policyinfo.Port
 			for _, port := range ports {
-				if port := r.convertToPolicyInfoPortForCIDRs(port); port != nil {
-					portList = append(portList, *port)
+				portInfo := r.convertToPolicyInfoPortForCIDRs(port)
+				if portInfo != nil {
+					portList = append(portList, *portInfo)
+				} else {
+					ports := r.getIngressRulesPorts(ctx, policy.Namespace, &policy.Spec.PodSelector, []networking.NetworkPolicyPort{port})
+					portList = append(portList, ports...)
```

Comment:
We should only support named ports for ingress. I think we need do same if condition check on `policyType` here as line 140.